### PR TITLE
N2O + Plasma = BZ + Cold

### PR DESCRIPTION
**Pull request by Cyberboss**
Magic numbers need to be made into defines but I want to get some feedback

:cl:
add: N2O and Plasma can now be combined to make BZ gas. Be careful, this is an endothermic reaction
/:cl:

### CODE REVIEW

**Comment by RandomMarine** [comment]
So does this make a viable (if messy) firefighting option?

**Comment by bgobandit** [comment]
as the porter of BZ I approve

**Comment by TribeOfBeavers** [comment]
This could make plasma setups for the SM impossible.

**Comment by bgobandit** [comment]
the SM needs a BZ reaction anyway

**Comment by Anonmare** [comment]
I hope BZ doesn't destabilise the SM. Maybe have it affect the SM the same way N2O does

**Comment by duncathan** [comment]
is there any reasoning behind its requirements or is it just a "we need a BZ reaction" sort of deal

**Comment by Cyberboss** [comment]
It's get the concept out there and then balance the numbers as appropriate

On May 11, 2017 4:55 PM, "duncathan salt" <notifications@github.com> wrote:

is there any reasoning behind its requirements or is it just a "we need a
BZ reaction" sort of deal

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/tgstation/tgstation/pull/27126#issuecomment-300914419>,
or mute the thread

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 1]
```diff
@@ -180,6 +180,34 @@
...
+	var/avail_heat = air.temperature - TCMB
+	
+	//3 degrees required per one tenth of a canister
+	var/max_moles = min(cached_gases["plasma"][MOLES], cached_gases["n2o"][MOLES], MOLES_CELLSTANDARD / 25, avail_heat / 0.03)	
+	
+	air.temperature -= max_moles * 0.03
+	air.assert_gas("bz")
+	cached_gases["bz"][MOLES] += max_moles * 1.5
+	cached_gases["plasma"][MOLES] -= max_moles
+	cached_gases["n2o"][MOLES] -= max_moles
```
Comment (duncathan) [inline code/modules/atmospherics/gasmixtures/reactions.dm]: Make sure to `garbage_collect()` after removing gases

**Review by duncathan — CHANGES_REQUESTED** [review]
I'm not sure yet if I agree with this quite the way it is but I agree that we should have a reaction to produce BZ. Let's talk about it for a bit before a decision gets made.

**Comment by ExcessiveUseOfCobblestone** [comment]
> This could make plasma setups for the SM impossible.

**Comment by YakumoChen** [comment]
this actually makes using BZ for the literally only one current useful purpose (slime stasis) unviable since slimes die from cold. You'd still have to order it from cargo to get any use from it, unless you can heat it back up.

File: code/modules/atmospherics/gasmixtures/reactions.dm (hunk 1 above)
Comment (Cyberboss) [inline code/modules/atmospherics/gasmixtures/reactions.dm]: The gas is already garbage collected in `gas_mixture/react()` if a reaction returns REACTING

**Comment by RandomMarine** [comment]
Science has a heater, y'know.

**Comment by Cyberboss** [comment]
@YakumoChen BZ by itself won't get cold. The cold comes from mixing plasma and N2O to create BZ

**Comment by KorPhaeron** [comment]
BZ is dumb

**Comment by duncathan** [comment]
kor makes a good point. we should make gases good before making them more accessible

**Pull request closed without merging**